File: engine-desktop/src/com/magma/engine/debug/ui/LabelUpdater.java
package com.magma.engine.debug.ui;

import com.badlogic.gdx.scenes.scene2d.Action;
import com.badlogic.gdx.scenes.scene2d.ui.Label;

public abstract class LabelUpdater extends Action {
	
	@Override
	public boolean act(float delta) {
		Label label = (Label) getActor();
		label.setText(updateText());
		return false;
	}
	
	public abstract String updateText();
}

File: engine/src/com/magma/engine/debug/ui/LabelUpdater.kt
package com.magma.engine.debug.ui

import com.badlogic.gdx.scenes.scene2d.Action
import com.badlogic.gdx.scenes.scene2d.ui.Label

/***
 * A helper Action for having labels automatically update
 */
abstract class LabelUpdater(private var updater: () -> String) : Action() {
    override fun act(delta: Float): Boolean {
        val label = getActor() as Label
        label.setText(updater.invoke())
        return false
    }
}